pkg/scalers/aws_cloudwatch_scaler.go

```diff
@@ -189,6 +206,12 @@ func parseAwsCloudwatchMetadata(config *scalersconfig.ScalerConfig) (*awsCloudwa
 	}
 	meta.minMetricValue = minMetricValue
 
+	errorWhenMetricValuesEmpty, err := getBoolMetadataValue(config.TriggerMetadata, "errorWhenMetricValuesEmpty", false, false)
+	if err != nil {
+		return nil, err
+	}
+	meta.errorWhenMetricValuesEmpty = errorWhenMetricValuesEmpty
```

Comment:
Please, use the helper that we have for reading values: https://github.com/kedacore/keda/blob/main/pkg/scalers/scaler.go#L242

Currently, we are migrating all the scalers to this new and unified approach

Reply:
About the key name, in other scalers we use `ignoreNullValues`, can we align this parameter with other scalers?

Reply:
> About the key name, in other scalers we use ignoreNullValues, can we align this parameter with other scalers?

I went with `errorWhenNullValues`. Let me know your thoughts here, and if you are good with the name, I'll update the docs PR.

> Please, use the helper that we have for reading values: https://github.com/kedacore/keda/blob/main/pkg/scalers/scaler.go#L242

I changed this over for `errorWhenNullValues` only in this PR, I'll present a new PR that changes over the rest of the values for the AWS CloudWatch scaler.

Reply:
> I changed this over for errorWhenNullValues only in this PR, I'll present a new PR that changes over the rest of the values for the AWS CloudWatch scaler.

Yeah, I meant using the helper only for this new value 😄 

> I went with errorWhenNullValues. Let me know your thoughts here, and if you are good with the name, I'll update the docs PR.

I'd prefer using the same approach as other scalers for the sake of the consistence. @zbynek @tomkerkhove ?

Reply:
> Yeah, I meant using the helper only for this new value 😄

I had a run through of switching over the rest of the inputs, excluding auth, in this PR: https://github.com/kedacore/keda/pull/5639

Reply:
> I'd prefer using the same approach as other scalers for the sake of the consistence. @zroubalik @tomkerkhove ?

Agree

Reply:
Oh this my bad, I misread this ask -- I just read the prom scaler, let me fix that.